ndcube/utils/sequence.py

```diff
@@ -323,9 +332,16 @@ def _convert_cube_like_index_to_sequence_slice(cube_like_index, common_axis_cube
     """
     # Derive cumulative lengths of cubes along common axis.
     cumul_common_axis_cube_lengths = np.cumsum(common_axis_cube_lengths)
-    # If cube_like_index is within first cube in sequence, it is
+    # If cube_like_index is within 0th cube in sequence, it is
     # simple to determine the sequence and common axis indices.
-    if cube_like_index < cumul_common_axis_cube_lengths[0]:
+    try:
+        index_in_0th_cube = cube_like_index < cumul_common_axis_cube_lengths[0]
```

Comment:
What's returning `None` here? I feel like this could be done without the `try`

Reply:
The issue is in comparing ```cube_like_index``` with ```cumul_common_axis_cube_lengths[0]```.  The latter is always a number.  The former is also a number except in one circumstance.  That is when the slice object from which ```cube_like_index``` is derived has a negative step and a stop parameter of None, i.e. the last included element is the 0th.  In this case, the stop parameter cannot be expressed by a number since stop means "up to but not including", and ```-1``` has a different meaning in Python slicing.  Therefore, in this case comparing ```cube_like_index``` with ```cumul_common_axis_cube_lengths[0]``` will raise a ```TypeError``` complaining about comparing ```int``` to ```None``` which can only mean that the index is in the first cube and so the if statement below must be entered.